docs/OV_Runtime_UG/integrate_with_your_application.md

```diff
@@ -0,0 +1,280 @@
+
+        .. doxygensnippet:: docs/snippets/src/main.py
+           :language: python
+           :fragment: [part2_4]
+
+@endsphinxdirective
+
+The code above creates a compiled model associated with a single hardware device from the model object.
+It is possible to create as many compiled models as needed and use them simultaneously (up to the limitation of the hardware resources).
+To learn how to change the device configuration, read the [Query device properties](./supported_plugins/config_properties.md) article.
+
+### Step 3. Create an Inference Request
+
+`ov::InferRequest` class provides methods for model inference in the OpenVINO™ Runtime.
+This section demonstrates a simple pipeline, to get more information about other use cases, read the [InferRequest documentation](./ov_infer_request.md) dedicated article.
+Create an infer request using the following code:
+
+@sphinxdirective
+
+.. tab:: C++
+
+    .. doxygensnippet:: docs/snippets/src/main.cpp
+       :language: cpp
+       :fragment: [part3]
+
+.. tab:: Python
+
+    .. doxygensnippet:: docs/snippets/src/main.py
+       :language: python
+       :fragment: [part3]
+
+@endsphinxdirective
+
+### Step 4. Set Inputs
+
+You can use external memory to create `ov::Tensor` and use the `ov::InferRequest::set_input_tensor()` method to put this tensor on the device:
+
+@sphinxdirective
+
+.. tab:: C++
+
+    .. doxygensnippet:: docs/snippets/src/main.cpp
+       :language: cpp
+       :fragment: [part4]
+
+.. tab:: Python
+
+    .. doxygensnippet:: docs/snippets/src/main.py
+       :language: python
+       :fragment: [part4]
+
+@endsphinxdirective
+
+### Step 5. Start Inference
+
+OpenVINO™ Runtime supports inference in asynchronous or synchronous mode. Async API usage can improve overall frame-rate of the application, because rather than wait for inference to complete, the app can continue doing things on the host, while the accelerator is busy. You can use `ov::InferRequest::start_async()` to infer the model in the asynchronous mode and call `ov::InferRequest::wait()` to wait for the inference results:
+
+@sphinxdirective
+
+.. tab:: C++
+
+    .. doxygensnippet:: docs/snippets/src/main.cpp
+       :language: cpp
+       :fragment: [part5]
+
+.. tab:: Python
+
+    .. doxygensnippet:: docs/snippets/src/main.py
+       :language: python
+       :fragment: [part5]
+
+@endsphinxdirective
+
+The asynchronous mode supports two methods to get the inference results:
+  * `ov::InferRequest::wait_for()` - Waits until the specified timeout (in milliseconds) has elapsed or the inference result becomes available, whichever comes first.
+  * `ov::InferRequest::wait()` - Waits until the inference result becomes available.
+
+Both requests are thread-safe, which means they can be called from different threads without exposing erroneous behavior or producing unpredictable results.
+
+Multiple requests for single `CompiledModel` are executed sequentially one by one in the FIFO order.
```

Comment:
This statement is not right, let's remove it. Later we need to insert a link to perf guide